resources/splunk_service.rb

```diff
@@ -0,0 +1,109 @@
+# Cookbook Name:: cerner_splunk_ingredient
+# Resource:: splunk_service
+#
+# Resource for managing Splunk as a system service
+class SplunkService < ChefCompat::Resource
+  include CernerSplunk::PlatformHelpers, CernerSplunk::ResourceHelpers,
+          CernerSplunk::ServiceHelpers, CernerSplunk::RestartHelpers
+
+  resource_name :splunk_service
+
+  property :name, String, name_property: true, desired_state: false, identity: true
+  property :package, [:splunk, :universal_forwarder], required: true
+  property :user, String
+  property :ulimit, Integer
+
+  default_action :start
+
+  def after_created
+    package_from_name unless property_is_set? :package
+  end
+
+  def service_name
+    @service_name ||= service_names[package][node['os'].to_sym]
+  end
+
+  action_class do
+    def service_action(desired_action)
+      service service_name do
+        provider Chef::Provider::Service::Systemd if systemd_is_init?
+        supports start: true, stop: true, restart: true, status: true
+        action desired_action
+      end
+    end
+
+    def configure_service
+      # Use run_action to execute immediately and avoid a race condition with /etc/init.d/splunk not existing yet.
```

Comment:
Yeah, it's involved with the ulimit values.

Particularly in an action on the Linux provider, I have something like this:
```Ruby
configure_service

converge_if_changed :ulimit do
  write_initd_ulimit ulimit
end
```

configure_service should be calling enable boot-start and generating the init.d script in time for write_initd_ulimit (which writes the ulimit into the init.d script), but it does not. Calling run_action will execute the resource immediately, the same way you would call run_action to execute a resource at compile time instead of converge time.

I don't know why this happens exactly, but I suspect it has something to do with DSL evaluation, and converge_if_changed might play a part in that.